src/main/java/org/opensearch/knn/quantization/quantizer/Quantizer.java

```diff
@@ -0,0 +1,37 @@
+package org.opensearch.knn.quantization.quantizer;
+
+import org.opensearch.knn.quantization.models.quantizationOutput.QuantizationOutput;
+import org.opensearch.knn.quantization.models.quantizationState.QuantizationState;
+import org.opensearch.knn.quantization.models.requests.TrainingRequest;
+
+import java.io.IOException;
+
+/**
+ * The Quantizer interface defines the methods required for training and quantizing vectors
```

Comment:
If we wanted to do some kind of asymetric distance computation in the future, could we change this interface easily to do that?

Reply:
yes ..we cab have default here so that otheres are not affected.